### PR TITLE
Update bespin.amazon.s3 for boto3

### PR DESCRIPTION
**Pull request by atward**
Relates #17.

boto3 provides a lot of this lifting natively.

### CODE REVIEW

File: bespin/amazon/s3.py [hunk 1]
```diff
@@ -43,147 +46,39 @@ def s3_location(self, value):
...
-        log.info("Uploading from %s (%s) to %s in one go", source, humanize.naturalsize(source_size), destination_file.full)
+        dest = self.s3_location(destination_path)
 
-        bucket = self.get_bucket(destination_file.bucket)
-        key = Key(bucket)
-        key.name = destination_file.key
-        key.set_contents_from_filename(source_filename, policy='authenticated-read')
-        key.close()
+        log.info("Uploading from %s (%s) to %s", source, humanize.naturalsize(source_size), dest.full)
+        self.conn.upload_file(source, dest.bucket, dest.key[1:])
```
Comment (delfick) [inline bespin/amazon/s3.py:103]: I can't remember why I cared enough to upload "large" files in parts and I wonder if the need for that goes away with boto3 or not...

Also, I'm not sure why set policy='authenticated-read' in upload_file_to_s3_as_single

File: bespin/amazon/s3.py [hunk 2]
```diff
@@ -43,147 +46,39 @@ def s3_location(self, value):
...
-    def a_multipart_upload(self, bucket, key):
-        mp = None
-        try:
-            mp = bucket.initiate_multipart_upload(key)
-            yield mp
-        except:
-            if mp:
-                mp.cancel_upload()
-            raise
+        if key == '/':
```
Comment (atward) [inline bespin/amazon/s3.py]: I'll wrap this in BespinError()

File: bespin/amazon/s3.py (hunk 1 above)
Comment (atward) [reply bespin/amazon/s3.py:103]: You need to (or historically had to) use multipart uploads for files larger than a certain amount. Botocore handles that via boto3.s3.transfer which upload_file() wraps.

Comment (delfick) [reply bespin/amazon/s3.py:103]: right, seems I chose 30mb arbitrarily though https://github.com/delfick/bespin/commit/a7c4ec7ae45f0779294a150014e3772c97501b98

**Review by delfick — APPROVED** [review]
Yay for boto3 being nicer than boto :)

File: bespin/amazon/s3.py [hunk 3]
```diff
@@ -43,147 +48,42 @@ def s3_location(self, value):
         key = info.path
         return S3Location(bucket, key, value)
 
-    def copy_key(self, frm, to):
```
Comment (delfick) [inline bespin/amazon/s3.py]: can we keep this?

It's very possible that it's used outside of bespin

**Comment by delfick** [comment]
:+1:

**Pull request merged**